backend/src/main/java/com/htecgroup/skynest/service/impl/FileServiceImpl.java

```diff
@@ -172,6 +170,47 @@ public List<FileResponse> getAllFilesWithParent(UUID parentFolderId) {
     return asFileResponseList(allFiles);
   }
 
+  @Override
+  public void moveFileToFolder(UUID fileId, UUID destinationFolderId) {
+    FileMetadataEntity fileMetadataEntity = findFileMetaDataEntity(fileId);
+    FolderEntity folderEntity =
+        folderRepository.findById(destinationFolderId).orElseThrow(FolderNotFoundException::new);
+    checkIfFileAlreadyInsideFolder(fileMetadataEntity, folderEntity);
+    fileMetadataEntity.setParentFolder(folderEntity);
```

Comment:
The same can be done here as the moveToRoot method, this can be moveToFolder for example, and maybe that way you wont expose the setters